### PR TITLE
EN-12707: improve sc generation for system vm calls

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- the SCRs resulted after system vm calls were misleading (for example, when creating a new delegation contract) 
  
## Proposed Changes
- set `SenderAddress` for `outputTransfer` inside `eei.go` if the flag is enabled. This will help generate SCRs in the correct way 

## Testing procedure
Start a testnet and create a new delegation contract ([docs](https://docs.elrond.com/validators/delegation-manager/#creating-a-new-delegation-contract)). after this, you should see 3 SCRs:
- `erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqylllslmq6y6` -> `CONTRACT_ADDRESS` 1250 egld
- `CONTRACT_ADDRESS` -> `erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l` 1250 egld
- scr for gas refund (optional)
In comparison, here's an example on how the SCRs were generated before: [explorer](https://explorer.elrond.com/transactions/7f21cda7076dd401302e4336cedef5273be77ba2aa98153da83ed978ced75d52)

### CODE REVIEW

File: vm/systemSmartContracts/delegation_test.go [hunk 1]
```diff
@@ -285,12 +288,15 @@ func TestDelegationSystemSC_ExecuteNilArgsShouldErr(t *testing.T) {
...
-		&mock.RaterMock{})
+	eei, _ := NewVMContext(VMContextArgs{
+		BlockChainHook:                          &mock.BlockChainHookStub{},
+		CryptoHook:                              hooks.NewVMCryptoHook(),
+		InputParser:                             &mock.ArgumentParserMock{},
+		ValidatorAccountsDB:                     &stateMock.AccountsStub{},
+		ChanceComputer:                          &mock.RaterMock{},
+		EpochNotifier:                           &epochNotifier.EpochNotifierStub{},
+		SetSenderInEeiOutputTransferEnableEpoch: 0,
+	})
```
Comment (ssd04) [inline vm/systemSmartContracts/delegation_test.go:299]: can't we use `createDefaultEei()` here also?

File: integrationTests/vm/delegation/delegationScenarios_test.go [hunk 2]
```diff
@@ -31,6 +31,38 @@ import (
...
+	tpn.BlockchainHook.SetCurrentHeader(&block.MetaBlock{Nonce: 1})
+
+	// create new delegation contract
+	delegationScAddress := deployNewSc(t, tpn, maxDelegationCap, serviceFee, value, tpn.OwnAccount.Address)
+
+	scrsHandler, _ := tpn.InterimProcContainer.Get(block.SmartContractResultBlock)
+	scrs := scrsHandler.GetAllCurrentFinishedTxs()
+
+	numExpectedScrsFound := 0
+	// we expect 2 scrs:
```
Comment (ssd04) [inline integrationTests/vm/delegation/delegationScenarios_test.go:50]: is this valid?

File: vm/systemSmartContracts/delegation_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply vm/systemSmartContracts/delegation_test.go:299]: yes, fixed

File: integrationTests/vm/delegation/delegationScenarios_test.go (hunk 2 above)
Comment (bogdan-rosianu) [reply integrationTests/vm/delegation/delegationScenarios_test.go]: yes, changed to `we expect the following 2 scrs`, as there can be an additional scr (for refunded gas), but it is not in the scope of this test

**Review by ssd04 — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by sasurobert — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**